### ISSUE TITLE
Improvements upon eldoc box behavior

### ISSUE THREAD

**Issue by andreyorst** [casouri/eldoc-box#21]
This plugin is now the definitive way of using eldoc for me. But I think it can be improved further a bit.

First of, now popup moves as long as we type text, or move with directional shortcuts. This is nice when typing and really annoying when moving around the code, because box closes most of the code below the cursor, even when you've moved to a different location, where the popup no longer makes sense.

I think it is logical to make eldoc box follow the cursor only on typing events, eg when modifying text in any way, and quickly disappear if we leaving the line on which it was displayed. Also hiding eldoc-box by using `C-g` would be nice addition.

Another thing that's bothers me is that you have to enable two modes in order to make eldoc hover near the cursor:

```emacs-lisp
(eldoc-box-hover-mode)
(eldoc-box-hover-at-point-mode)
```

Why can't we simply call one of those? One for hover mode, the other for hover at point mode.

**Comment by casouri** [comment]
> I think it is logical to make eldoc box follow the cursor only on typing events, eg when modifying text in any way, and quickly disappear if we leaving the line on which it was displayed. Also hiding eldoc-box by using C-g would be nice addition.

Good point! Btw at-point mode already have the C-g feature, it doesn’t work?

> One for hover mode, the other for hover at point mode.

Good idea.

**Comment by casouri** [comment]
I do have the intention to improve the responsiveness of Eldoc-box, but I don’t have time right now. That is a long-term goal.

**Comment by casouri** [comment]
Try [this branch](https://github.com/casouri/eldoc-box/tree/new-feature) and see if there's a problem or space for improvement.

What you should see is moving point now hides childframe and typing keep the display in eldoc-box-hover-at-point-mode.

Also the two modes are not exclusive independent modes.

**Comment by andreyorst** [comment]
I've tried this branch and it seem to work slick! I've noticet this bug though, but I suspect that this is because I'm on gnome:

![image](https://user-images.githubusercontent.com/19470159/66112152-fce0e200-e5d2-11e9-9e81-8b4419889ec5.png)

eldoc-box is displayed above company. Gonna try this on a different WM to make sure.

**Comment by andreyorst** [comment]
No, this reproduces in xfce4. I guess I was to early on #15

**Comment by casouri** [comment]
I can't reproduce on KDE plasma. Could you try reproduce it with minimal config? I.e., `emacs -q` and only load company and eldoc-box.

**Comment by andreyorst** [comment]
Weird, but I also can't reproduce it with the minimal configuration..

**Comment by casouri** [comment]
That’s means your config has some culprit in it :)

**Comment by andreyorst** [comment]
Seems so. I have no complains so if you satisfied with the changes and code I think this is fixed and can be merged
